File: isletimsistemlericalismasiJava-calismalarim/app/src/main/java/com/example/isletimsistemicalismasi/uygulamalar/telefon/TelefonFragment.java
package com.example.isletimsistemicalismasi.uygulamalar.telefon;

import android.Manifest;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;

import androidx.core.app.ActivityCompat;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;

import com.example.isletimsistemicalismasi.databinding.FragmentTelefonBinding;


public class TelefonFragment extends Fragment {

    FragmentTelefonBinding binding;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentTelefonBinding.inflate(inflater,container,false);
                if (ActivityCompat.checkSelfPermission(requireContext(), Manifest.permission.CALL_PHONE) != PackageManager.PERMISSION_GRANTED) {

                    ActivityCompat.requestPermissions(requireActivity(), new String[]{Manifest.permission.CALL_PHONE}, 101);
                }
                binding.btnAra.setOnClickListener(view-> {
                    Intent callIntent = new Intent(Intent.ACTION_CALL);
                    callIntent.setData(Uri.parse("tel:" + binding.etNumara.getText().toString().trim()));
                    startActivity(callIntent);
                });

        return binding.getRoot();
    }
}

File: isletimsistemlericalismasikotlin-calismalarim/app/src/main/java/com/example/isletimsistemlericalismasikotlin/uygulamalar/telefon/TelefonFragment.kt
package com.example.isletimsistemlericalismasikotlin.uygulamalar.telefon

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.net.Uri
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.app.ActivityCompat
import com.example.isletimsistemlericalismasikotlin.R
import com.example.isletimsistemlericalismasikotlin.databinding.FragmentTelefonBinding

class TelefonFragment : Fragment() {

    private var _binding: FragmentTelefonBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        _binding = FragmentTelefonBinding.inflate(inflater, container, false)

        if (ActivityCompat.checkSelfPermission(
                requireContext(), Manifest.permission.CALL_PHONE
            ) != PackageManager.PERMISSION_GRANTED
        ) {
            ActivityCompat.requestPermissions(
                requireActivity(), arrayOf(Manifest.permission.CALL_PHONE), 101
            )
        }

        binding.btnAra.setOnClickListener {
            val callIntent = Intent(Intent.ACTION_CALL)
            callIntent.data = Uri.parse("tel:${binding.etNumara.text.toString().trim()}")
            startActivity(callIntent)
        }

        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
